File: scripts/generateOgImage.js
import { createCanvas } from 'canvas';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

const WIDTH = 1200;
const HEIGHT = 630;

const canvas = createCanvas(WIDTH, HEIGHT);
const ctx = canvas.getContext('2d');

// Background gradient
const gradient = ctx.createLinearGradient(0, 0, WIDTH, HEIGHT);
gradient.addColorStop(0, '#4f46e5');
gradient.addColorStop(1, '#7c3aed');
ctx.fillStyle = gradient;
ctx.fillRect(0, 0, WIDTH, HEIGHT);

// Decorative circles
ctx.fillStyle = 'rgba(255, 255, 255, 0.05)';
ctx.beginPath();
ctx.arc(100, 100, 200, 0, Math.PI * 2);
ctx.fill();
ctx.beginPath();
ctx.arc(1100, 530, 250, 0, Math.PI * 2);
ctx.fill();

// Title
ctx.fillStyle = '#ffffff';
ctx.font = 'bold 72px sans-serif';
ctx.textAlign = 'center';
ctx.fillText('Chinese Name Generator', WIDTH / 2, 220);

// Subtitle
ctx.font = '36px sans-serif';
ctx.fillStyle = 'rgba(255, 255, 255, 0.9)';
ctx.fillText('Get Your Perfect Chinese Name', WIDTH / 2, 300);

// Features
ctx.font = '28px sans-serif';
ctx.fillStyle = 'rgba(255, 255, 255, 0.8)';
const features = ['Personalized Names', 'Zodiac Analysis', 'Five Elements (Wu Xing)', 'Pronunciation Guide'];
features.forEach((text, i) => {
  ctx.fillText(text, WIDTH / 2, 380 + i * 50);
});

// URL
ctx.font = '24px sans-serif';
ctx.fillStyle = 'rgba(255, 255, 255, 0.6)';
ctx.fillText('chinese-name.m9ai.work', WIDTH / 2, 600);

// Save
const buffer = canvas.toBuffer('image/png');
const outputPath = path.join(__dirname, '..', 'public', 'og-image.png');
fs.writeFileSync(outputPath, buffer);
console.log('OG image generated:', outputPath);

File: scripts/generateOgImage.py
from PIL import Image, ImageDraw, ImageFont
import os

WIDTH, HEIGHT = 1200, 630

img = Image.new('RGB', (WIDTH, HEIGHT), color='#4f46e5')
draw = ImageDraw.Draw(img)

# Gradient background (simulated with rectangles)
for y in range(HEIGHT):
    r = int(79 + (124 - 79) * y / HEIGHT)
    g = int(70 + (58 - 70) * y / HEIGHT)
    b = int(229 + (237 - 229) * y / HEIGHT)
    draw.line([(0, y), (WIDTH, y)], fill=(r, g, b))

# Decorative circles
draw.ellipse([-100, -100, 300, 300], fill=(255, 255, 255, 13))
draw.ellipse([900, 380, 1400, 880], fill=(255, 255, 255, 13))

# Try to load a nice font, fallback to default
try:
    title_font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", 72)
    subtitle_font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial.ttf", 36)
    feature_font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial.ttf", 28)
    url_font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial.ttf", 24)
except:
    title_font = ImageFont.load_default()
    subtitle_font = ImageFont.load_default()
    feature_font = ImageFont.load_default()
    url_font = ImageFont.load_default()

# Title
title = "Chinese Name Generator"
bbox = draw.textbbox((0, 0), title, font=title_font)
title_w = bbox[2] - bbox[0]
draw.text(((WIDTH - title_w) / 2, 180), title, font=title_font, fill=(255, 255, 255))

# Subtitle
subtitle = "Get Your Perfect Chinese Name"
bbox = draw.textbbox((0, 0), subtitle, font=subtitle_font)
sub_w = bbox[2] - bbox[0]
draw.text(((WIDTH - sub_w) / 2, 280), subtitle, font=subtitle_font, fill=(230, 230, 255))

# Features
features = ["Personalized Names", "Zodiac Analysis", "Five Elements (Wu Xing)", "Pronunciation Guide"]
start_y = 360
for i, text in enumerate(features):
    bbox = draw.textbbox((0, 0), text, font=feature_font)
    tw = bbox[2] - bbox[0]
    draw.text(((WIDTH - tw) / 2, start_y + i * 50), text, font=feature_font, fill=(200, 200, 255))

# URL
url = "chinese-name.m9ai.work"
bbox = draw.textbbox((0, 0), url, font=url_font)
uw = bbox[2] - bbox[0]
draw.text(((WIDTH - uw) / 2, 580), url, font=url_font, fill=(170, 170, 230))

# Save
output = os.path.join(os.path.dirname(__file__), '..', 'public', 'og-image.png')
img.save(output)
print(f"OG image saved to: {output}")
